src/App.tsx

```diff
@@ -1,14 +1,49 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { GroupBy } from './types/GroupBy';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [visibleTodos, setVisibleTodos] = useState(todos);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(todos[0]);
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    getTodos().then(response => {
+      setTodos(response);
+      setVisibleTodos(response);
+    }).finally(() => setIsLoading(false));
```

Comment:
Let’s use async await. It gives you a simpler way to work with asynchronous promise-based code